### ISSUE TITLE
Wrong error "expected expression" in parsing of braced-init-list inside template argument list

### ISSUE THREAD

**Issue by AndreyG** [llvm/llvm-project#58434]
Repro (https://gcc.godbolt.org/z/ehMnoqf7h):
```cpp
template<int>
void f();

void test() {
  f<{42}>();
}
```

**Comment by shafik** [comment]
This is [defect report 2450: braced-init-list as a template-argument](https://www.open-std.org/jtc1/sc22/wg21/docs/cwg_active.html#2450) which says:

>Since non-type template parameters can now have class types, it would seem to make sense to allow a braced-init-list as a template-argument, but the grammar does not permit it.

It is still active but has drafting status: https://www.open-std.org/jtc1/sc22/wg21/docs/cwg_index.html

Given NTTP it seems like we should allow at least some cases of `braced-init-list` for a `template-argument`, let me see if I can dig into the status more.

**Comment by cor3ntin** [comment]
Should be fixed by https://github.com/llvm/llvm-project/pull/73103
